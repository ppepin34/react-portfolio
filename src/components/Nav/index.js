import React from "react";

function Nav() {
    return (
        <header>
            <h2>
                <a data-testid='link' href="/">
                    Phil Pepin
                </a>
            </h2>
            <nav>
                <ul>
                    <li>
                        <a data-testid='about' href="#about">
                            About Me
                        </a>
                    </li>
                    <li>
                        <span>Contact</span>
                    </li>
                    <li>
                            <span>Portfolio</span>
                    </li>
                    <li>
                        <span>Resume</span>
                    </li>
                </ul>
            </nav>
        </header>
    )
}

export default Nav;