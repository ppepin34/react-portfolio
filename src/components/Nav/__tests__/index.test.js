import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from "..";

afterEach(cleanup);

describe('Nav component', () => {
    it('renders', () => {
        render(<Nav />);
    });

    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Nav />);
        expect(asFragment()).toMatchSnapshot();
    });

    describe('links are visible', () => {
        it('inserts text into the links', () => {
            const { getByTestId } = render(<Nav />);
            expect(getByTestId('link')).toHaveTextContent('Phil Pepin');
            expect(getByTestId('about')).toHaveTextContent('About Me');
        });
    })
})