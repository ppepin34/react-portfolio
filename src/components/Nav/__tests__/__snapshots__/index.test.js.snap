// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        Phil Pepin
      </a>
    </h2>
    <nav>
      <ul>
        <li>
          <a
            data-testid="about"
            href="#about"
          >
            About Me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li>
          <span>
            Portfolio
          </span>
        </li>
        <li>
          <span>
            Resume
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
